pkg/utils/utils_test.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestGetDomainName(t *testing.T) {
+	tests := []struct {
+		descr      string
+		resolvConf string
+		domainName string
+		shouldFail bool
+	}{
+		{
+			descr: "all good",
+			resolvConf: `
+nameserver 1.1.1.1
+search default.svc.abc.com svc.abc.com abc.com
+options ndots:5
+`,
+			domainName: "abc.com",
+			shouldFail: false,
+		},
+		{
+			descr: "missing search line",
+			resolvConf: `
+nameserver 1.1.1.1
+options ndots:5
+`,
+			domainName: "",
+			shouldFail: true,
+		},
+		{
+			descr: "non k8s resolv.conf format",
+			resolvConf: `
+nameserver 1.1.1.1
+search  abc.com xyz.com
+options ndots:5
+`,
+			domainName: "",
+			shouldFail: true,
+		},
+	}
+	for _, tt := range tests {
+		dn, err := getClusterDomainName(strings.NewReader(tt.resolvConf))
+		if err != nil {
```

Comment:
```
if got, want := (err != nil), t.shouldFail; got != want {
   t.Errorf("....")
} else if err == nil {
  if got, want := domain, t.domainName; got != want {
    t.Errorf("... domain: got: %s, want: %s",...)
  }
}
```

Reply:
In this sequence, it does not catch directly func returns err == nil but the expectation was to fail. Do you think it is sufficient to catch this condition just on domain name mismatch? Currently err==nil  validation covers 2 scenarios, 1 incorrect parsing of domain name when func returns nil and wrong domain name and 2 not detecting corruption of resolv.conf file. Please let me know if this explanation makes sense to you. Appreciate a lot your feedback.

Reply:
` it does not catch directly func returns err == nil but the expectation was to fail` -- wouldn't `(err != nil) != t.shouldFail` condition catch that? 
We want to make sure errors are returned when it's an error condition. But since your tests don't discriminate errors, just making sure error was returned in the erroneous condition will suffice. That's what my suggested change does.
Hope, this makes sense :)